src/assets/components/ViewArticle.jsx

```diff
@@ -1,23 +1,34 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById, getCommentByArticleId } from "../../utils";
+import { getArticleById, updateArticleVotes } from "../../utils";
+import CommentCard from "./CommentCard";
 
 export default function ViewArticle() {
     const {article_id} = useParams()
     const [article, setArticle] = useState({})
-    const [comments, setComments] = useState([])
 
     useEffect(() => {
         getArticleById(article_id).then((response) => {
             setArticle(response.article)
         })
     }, [article_id])
 
-    useEffect(() => {
-        getCommentByArticleId(article_id).then((response) => {
-            setComments(response.comments)
-        })
-    })
+    function increaseVote() {
```

Comment:
We could have these two functions be dynamic and take a +1 or -1 value through from the button value so we can make this more DRY.

Also in order to achieve this process we shouldn't be refetching the article from the API. This could cause problems if a lot of people are clicking this button at the same time. I could click upvote and the number goes up by hundreds. Can you remember how we handled this previously in the lecture?